### PR TITLE
feat(kinesisanalytics-flink): support Apache Flink 1.20

### PR DESCRIPTION
**Pull request by mazyu36**
### Issue # (if applicable)

N/A

### Reason for this change
Support new runtime.

Ref: https://aws.amazon.com/about-aws/whats-new/2024/09/amazon-managed-service-apache-flink-1-20/

### Description of changes
* Add Flink 1.20 to enum
* Improve unit tests and integ test to test all runtime versions

### Description of how you validated changes
Ran improved unit tests and integ tests.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-kinesisanalytics-flink-alpha/test/application.test.ts [hunk 1]
```diff
@@ -21,21 +21,32 @@ describe('Application', () => {
...
+    flink.Runtime.FLINK_1_8,
+    flink.Runtime.FLINK_1_11,
+    flink.Runtime.FLINK_1_13,
+    flink.Runtime.FLINK_1_15,
+    flink.Runtime.FLINK_1_18,
+    flink.Runtime.FLINK_1_19,
+    flink.Runtime.FLINK_1_20,
+  ];
+
+  test.each(flinkRuntimes)('Flink Application with %s', (runtime) => {
```
Comment (mazyu36) [inline packages/@aws-cdk/aws-kinesisanalytics-flink-alpha/test/application.test.ts:40]: Improve to test all runtimes

File: packages/@aws-cdk/aws-kinesisanalytics-flink-alpha/test/integ.application.lit.ts [hunk 2]
```diff
@@ -1,22 +1,36 @@
...
+  flink.Runtime.FLINK_1_8,
+  flink.Runtime.FLINK_1_11,
+  flink.Runtime.FLINK_1_13,
+  flink.Runtime.FLINK_1_15,
+  flink.Runtime.FLINK_1_18,
+  flink.Runtime.FLINK_1_19,
+  flink.Runtime.FLINK_1_20,
+];
+
+flinkRuntimes.forEach((runtime) => {
```
Comment (mazyu36) [inline packages/@aws-cdk/aws-kinesisanalytics-flink-alpha/test/integ.application.lit.ts:21]: Improved to test all runtimes

File: packages/@aws-cdk/aws-kinesisanalytics-flink-alpha/README.md [hunk 3]
```diff
@@ -59,7 +59,7 @@ snapshotting, monitoring, and parallelism.
 declare const bucket: s3.Bucket;
 const flinkApp = new flink.Application(this, 'Application', {
   code: flink.ApplicationCode.fromBucket(bucket, 'my-app.jar'),
-  runtime: flink.Runtime.FLINK_1_19,
+  runtime: flink.Runtime.FLINK_1_20,
```
Comment (lpizzinidev) [inline packages/@aws-cdk/aws-kinesisanalytics-flink-alpha/README.md:62]: Can you update row 83 as well for consistency?

**Review by lpizzinidev — APPROVED** [review]
👍

File: packages/@aws-cdk/aws-kinesisanalytics-flink-alpha/README.md (hunk 3 above)
Comment (mazyu36) [reply packages/@aws-cdk/aws-kinesisanalytics-flink-alpha/README.md:62]: Thanks. Updated.

**Review by GavinZZ — APPROVED** [review]
Love it, thanks!

**Comment by mazyu36** [comment]
@Megifyio update

**Comment by mazyu36** [comment]
@mergify update

**Comment by mazyu36** [comment]
@Mergifyio update

**Review by GavinZZ — APPROVED** [review]

**Pull request merged**